resources/public/pamela.ebnf

```diff
@@ -18,20 +18,25 @@ inherit = <INHERIT> <LB> symbol+ <RB>
 
 modes = <MODES> ( mode-enum | mode-map )
 mode-enum = <LB> keyword* <RB>
-mode-map = <LM> mode-init+ <RM>
+mode-map = <LM> mode-init* <RM>
 mode-init = keyword cond-expr
 cond-expr = ( TRUE | FALSE | and-expr | or-expr | implies-expr |
-              not-expr | equal-expr | call-expr )
+              not-expr | call-expr | equal-expr | same-expr | propositions-expr )
 and-expr = <LP> <AND> ( cond-expr | cond-operand )+ <RP>
 or-expr = <LP> <OR> ( cond-expr | cond-operand )+ <RP>
 implies-expr = <LP> <IMPLIES> ( cond-expr | cond-operand )+ <RP>
 not-expr = <LP> <NOT> ( cond-expr | cond-operand )+ <RP>
 equal-expr = <LP> <EQUAL> cond-operand+ <RP>
+same-expr = <LP> <SAME> cond-operand+ <RP>
 call-expr = <LP> <CALL> cond-operand+ <RP>
+propositions-expr = <LP> <PROPOSITIONS> <LB> prop-search+ <RB> (<WHERE> cond-expr)? <RP>
+prop-search = <LP> search-constraints? keyword cond-operand+ <RP>
+search-constraints = <LB> (wm  | ltm | <LP> recency <RP>)* <RB>
 cond-operand = ( literal | mode-ref | symbol-ref )
 mode-ref = <LP> <MODE_OF> symbol-ref keyword <RP>
+recency = <':recency'> number-ref
```

Comment:
All of the literals for reserved symbols are declared at the bottom of the ebnf file.  This should probably also be done for `:recency`, indirecting via something like <RECENCY>